snuba/util.py

```diff
@@ -55,21 +64,34 @@ def string_col(dataset, col):
         return 'toString({})'.format(escape_col(col))
 
 
-def escape_col(col):
-    if not col:
-        return col
-    elif SAFE_COL_RE.match(col):
-        # Column is safe to use without wrapping.
-        return col
+def escape_expression(expr: Optional[str], regex) -> Optional[str]:
+    if not expr:
+        return expr
+    elif regex.match(expr):
+        # Column/Alias is safe to use without wrapping.
+        return expr
     else:
-        # Column needs special characters escaped, and to be wrapped with
+        # Column/Alias needs special characters escaped, and to be wrapped with
         # backticks. If the column starts with a '-', keep that outside the
         # backticks as it is not part of the column name, but used by the query
         # generator to signify the sort order if we are sorting by this column.
-        col = ESCAPE_COL_RE.sub(r"\\\1", col)
+        col = ESCAPE_COL_RE.sub(r"\\\1", expr)
         return u'{}`{}`'.format(*NEGATE_RE.match(col).groups())
 
 
+def escape_alias(alias):
+    """
+    Alias escaping is different than column names when we introduce table aliases.
+    Using the column escaping function would consider "." safe, which is not for
+    an alias.
+    """
```

Comment:
I'm torn on whether this comment should be closer to the regular expression definition or something.